### PR TITLE
Update check box styles for ActionList to match the rest of Primer

### PR DESCRIPTION
**Pull request by camertron**
### What are you trying to accomplish?

This PR updates the check box styles for `ActionList` to match the rest of Primer in order to fix a rendering issue on non-retina screens. See the associated issue for more information.

### Screenshots

|Before|After|
|-|-|
|A screenshot of a SelectPanel component with smaller, older-style check boxes visible|A screenshot of a SelectPanel component with larger, newer-style check boxes visible|

### Integration

No changes necessary in production.

#### List the issues that this change affects.

Fixes https://github.com/github/primer/issues/4450

Closes # (type the GitHub issue number after #)

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### What approach did you choose and why?

As recommended in Slack, I copy/pasted the check box styles from the `Checkbox` component. Unfortunately I couldn't re-use them because the markup is different between `ActionList`s and `Checkbox`es.

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

### Merge checklist

~- [ ] Added/updated tests~
~- [ ] Added/updated documentation~
~- [ ] Added/updated previews (Lookbook)~
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

File: app/components/primer/alpha/action_list.pcss [hunk 1]
```diff
@@ -448,64 +448,31 @@ nav-list {
...
   &[aria-selected='true'] {
-    /* multiselect checkmark */
-    & .ActionListItem-multiSelectCheckmark {
-      visibility: visible;
-      opacity: 1;
-      transition:
-        visibility 0 linear 0,
-        opacity 50ms;
+    & .FormControl-checkbox {
+      background: var(--control-checked-bgColor-rest, var(--color-accent-fg));
```
Comment (langermank) [inline app/components/primer/alpha/action_list.pcss:452]:
```suggestion
      background: var(--control-checked-bgColor-rest);
```

File: app/components/primer/alpha/action_list.pcss [hunk 2]
```diff
@@ -448,64 +448,31 @@ nav-list {
...
+      border-color: var(--control-checked-borderColor-rest, var(--color-accent-fg));
```
Comment (langermank) [inline app/components/primer/alpha/action_list.pcss:453]:
```suggestion
      border-color: var(--control-checked-borderColor-rest);
```

**Review by langermank — APPROVED** [review]
Just for safety I would probably do a quick search in dotcom to see if anyone is overriding the classes removed here 😄

**Comment by camertron** [comment]
Did a search in dotcom for these classes. It appears they are not used 🎉

**Pull request merged**